beacon-chain/rpc/helpers/broadcast.go

```diff
@@ -0,0 +1,173 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/core/peerdas"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/blocks"
+	ethpb "github.com/OffchainLabs/prysm/v6/proto/prysm/v1alpha1"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+// DataColumnBroadcastFunc is a function for broadcasting data column sidecars.
+type DataColumnBroadcastFunc func(root [32]byte, subnet uint64, sidecar *ethpb.DataColumnSidecar) error
+
+// BlobBroadcastFunc is a function for broadcasting blob sidecars.
+type BlobBroadcastFunc func(ctx context.Context, index uint64, sidecar *ethpb.BlobSidecar) error
+
+// DataColumnOption configures options for broadcasting data column sidecars.
+type DataColumnOption func(*dataColumnOptions)
+
+type dataColumnOptions struct {
+	onReceiveDataColumns   func(columns []blocks.VerifiedRODataColumn) error
+	onDataColumnsProcessed func(columns []blocks.VerifiedRODataColumn)
+}
+
+// WithDataColumnReceiver sets the callback for receiving data columns.
+func WithDataColumnReceiver(fn func(columns []blocks.VerifiedRODataColumn) error) DataColumnOption {
+	return func(o *dataColumnOptions) {
+		o.onReceiveDataColumns = fn
+	}
+}
+
+// WithDataColumnProcessedCallback sets the callback to be called after data columns are processed.
+func WithDataColumnProcessedCallback(fn func(columns []blocks.VerifiedRODataColumn)) DataColumnOption {
+	return func(o *dataColumnOptions) {
+		o.onDataColumnsProcessed = fn
+	}
+}
+
+// BroadcastDataColumnSidecars broadcasts data column sidecars concurrently and optionally receives them.
+func BroadcastDataColumnSidecars(
```

Comment:
I think a warning should be added in the godoc of this function because it transforms sidecars into verified sidecars.
Like:
"WARNING: This function should be called only with trusted sidecars, since it will upgrade them as verified without any further check."

Same for the `BroadcastBlobSidecars`.
Maybe Kasey could have a better idea.